### PR TITLE
Media Library: show 200 GB storage limit for Business and eCommerce sites

### LINKED ISSUE #35965: Show storage limit and usage for Business and eCommerce sites
The storage space limit and usage indicator is not currently shows on Business and eCommerce sites (both simple and Atomic). We need to expose this to display the upload limit and current usage.

#### Steps to reproduce
1. Starting at URL: https://wordpress.com/media/site
2. Note the indicator is not shown on Business and eCommere sites as it is on lower plans

#### What I expected
To see how much space I am using and my space limit.

#### What happened instead
I cannot see the space bar.

#### Screenshot / Video
Here is what this looks like on a Premium site (the upsell would not be present on these sites as they are on our maximum storage plans):

Screen Shot 2019-09-03 at 10 48 23 PM

### PR DESCRIPTION
**Pull request by travisw**
#### Changes proposed in this Pull Request

Currently, Business and eCommerce plans do not show users how much of their allotted quota they have used in the media library.

This change will now show that info by updating the features in those plans to use a new `FEATURE_200GB_STORAGE` constant, instead of `FEATURE_UNLIMITED_STORAGE`, which no longer makes sense since they don't have unlimited storage.

This should work for both Simple and Atomic sites.

[Previously we had just updated the copy](https://github.com/Automattic/wp-calypso/commit/9083014a8c07b16449f1b961d28c25b2f74ae020) that's displayed for the unlimited space feature to show 200GB. This change makes the storage space feature for Business and eCommerce more inline with the other plans.

Relies on API endpoint update here: D40523-code

More info: pau2Xa-Xd-p2

Fixes #35965

**TODO:**
* ~~Fix tests.~~
* ~~Make sure the "upgrade" link still shows only when intended.~~

-------------

#### Testing instructions

Apply this wpcom patch on the back end: D40523-code

Boot up this branch locally.

Sandbox `public-api.wordpress.com`.

You'll need a few test sites.

* Business (Simple site).
* eCommerce (Simple site).
* Business or eCommerce (Atomic site).
* Non-Business-or-eCommerce (Premium, Personal, or Free).
* Jetpack site.

Business (Simple site)
* Visit the media library (http://calypso.localhost:3000/media/YOUR_DOMAIN).
* You should see "X% of 200 GB used", **without** an "Upgrade" link.
Screen Shot 2020-03-19 at 1 27 42 PM

eCommerce (Simple site)
* Visit the media library (http://calypso.localhost:3000/media/YOUR_DOMAIN).
* You should see "X% of 200 GB used", **without** an "Upgrade" link.
Screen Shot 2020-03-19 at 1 27 42 PM

Business or eCommerce (Atomic site)
* Visit the media library (http://calypso.localhost:3000/media/YOUR_DOMAIN).
* You should see "X% of 200 GB used", **without** an "Upgrade" link.
Screen Shot 2020-03-19 at 1 27 42 PM

Non-Business-or-eCommerce (Premium, Personal, or Free)
* Visit the media library (http://calypso.localhost:3000/media/YOUR_DOMAIN).
* You should see "X% of 6 GB used" **with** an "Upgrade" link (6 GB for Personal, or the [correct storage space limit](https://wordpress.com/pricing/) for the plan you're testing).
Screen Shot 2020-03-18 at 4 02 04 PM

Jetpack site
* Visit the media library (http://calypso.localhost:3000/media/YOUR_DOMAIN).
* You **should not** see the storage space messaging.
Screen Shot 2020-03-18 at 4 06 16 PM

### CODE REVIEW

**Comment by lancewillett** [comment]
Happy to test this in the UI once it's ready for review.

Note: your description "Relies on API endpoint update here..." didn't show a link for me.

**Comment by travisw** [comment]
> Happy to test this in the UI once it's ready for review.
> 
> Note: your description "Relies on API endpoint update here..." didn't show a link for me.

Thanks for the quick response Lance! Was just updating the description and test plan, stay tuned.

**Comment by lancewillett** [comment]
Testing with the live branch, using Safari, Firefox, and Chrome (all on macOS Catalina, desktop).

> Business (Simple site).

business-simple

I don't see any message in Media UI in Calypso.

WP Admin shows `You are currently using 12.6 MB out of 200.0 GB upload limit (0%).`

> eCommerce (Simple site).

Didn't test this case, as I'm not sure how to get a site like this.

> Business or eCommerce (Atomic site).

business-atomic-no-msg

Business, Atomic — I don't see any message (nor WP Admin).

ecommerce-atomic

eCommerce, Atomic — I don't see any message (nor WP Admin).

Screen Shot 2020-03-20 at 11 11 43

Business, Atomic on `blatomic.com` - I see `0% of 1 TB used` (no message in WP Admin).

> Non-Business-or-eCommerce (Premium, Personal, or Free).

free-site

Existing site, Free plan, I see `1.4% of 3 GB used` and not 6 GB as your testing note indicated.

premium

New site, Premium plan, I see `0% of 13 GB used`

> Jetpack site.

Good to go on Jetpack sites, no message visible in Calypso "Media" UI. 👍

**Comment by travisw** [comment]
Thanks @lancewillett . For this PR to work properly it needs the backend patch applied and for the public API to be sandboxed. So I _don't think_ the live branch would work.
> Apply this wpcom patch on the back end: D40523-code

I forgot to mention the sandboxing part in the test plan. I'll update that.

**Comment by travisw** [comment]
> Existing site, Free plan, I see `1.4% of 3 GB used` and not 6 GB as your testing note indicated.

That should be correct since you tested with a free plan. My example used the Personal plan.

**Comment by jaswrks** [comment]
> Business (Simple site)
> You should see "X% of 200 GB used", without an "Upgrade" link.

✅ 

Screen Shot 2020-03-20 at 17 41 11

**Comment by jaswrks** [comment]
> eCommerce (Simple site)
> You should see "X% of 200 GB used", without an "Upgrade" link.

✅ Tested via SA by upgrading to eCommerce, but non-Atomic.

Screen Shot 2020-03-20 at 18 36 30

**Comment by jaswrks** [comment]
> Business or eCommerce (Atomic site)
> You should see "X% of 200 GB used", without an "Upgrade" link.

✅ Tested an Atomic Business site.

Screen Shot 2020-03-20 at 18 38 15

**Comment by jaswrks** [comment]
> Non-Business-or-eCommerce (Premium, Personal, or Free)
> You should see "X% of 6 GB used" with an "Upgrade" link (6 GB for Personal, or the correct storage space limit for the plan you're testing).

✅ Free, Personal, Premium

Screen Shot 2020-03-20 at 18 40 10

Screen Shot 2020-03-20 at 18 41 32

Screen Shot 2020-03-20 at 18 44 10

**Comment by jaswrks** [comment]
> Jetpack site
> You should not see the storage space messaging.

✅ Jetpack site looks good for me also.

Screen Shot 2020-03-20 at 18 46 40

**Comment by jaswrks** [comment]
@travisw Special case for a P2 on the Business plan? Shows **6 GB**.

Screen Shot 2020-03-20 at 18 41 32 (1)

File: client/lib/plans/features-list.js [hunk 1]
```diff
@@ -587,6 +587,18 @@ export const FEATURES_LIST = {
 			i18n.translate( 'Upload more images, videos, audio, and documents to your website.' ),
 	},
 
+	[ constants.FEATURE_200GB_STORAGE ]: {
+		getSlug: () => constants.FEATURE_200GB_STORAGE,
+		getTitle: () =>
+			i18n.translate( '{{strong}}200 GB{{/strong}} Storage Space', {
```
Comment (jaswrks) [inline client/lib/plans/features-list.js]: @travisw How should we test this? Are the strings already translated elsewhere?

**Comment by lancewillett** [comment]
Re-tested with the backend patch D40523-code applied on my sandbox, sandboxing the WP.com REST API.

Results now match the test plan.

✅ Business (Simple site)
✅ eCommerce (Simple site)
✅ Business or eCommerce (Atomic site)
✅ Non-Business-or-eCommerce (Premium, Personal, or Free)
✅ Jetpack site

File: client/lib/plans/features-list.js (hunk 1 above)
Comment (travisw) [reply client/lib/plans/features-list.js]: Yep, this should already be translated. It's pulled from here:
https://github.com/Automattic/wp-calypso/blob/448191032ddf297ff5803cdc6648b11c73307511/client/lib/plans/features-list.js#L288

**Comment by travisw** [comment]
> @travisw Special case for a P2 on the Business plan? Shows **6 GB**.

Interesting, thanks for spotting that. I've seen some incorrect result on some sites under my account. I'm guessing this has something to do with the various ways that we might add a plan subscription to a site, that might be different than how a user would go about it.

I'll look into it.

**Pull request merged**